app/adapters/alma_adapter.rb

```diff
@@ -103,4 +107,59 @@ def perm_location(bib_item)
     def apikey
       Rails.configuration.alma[:read_only_apikey]
     end
+
+    # @param mms_id [String]. e.g id = "991227850000541"
+    # @param all_holding_ids [Array<String>]
+    # @return [Hash<String, Array>] hash with holding id as key and notes as values
+    def bib_item_holding_notes(mms_id, all_holding_ids)
+      notes_from_bib_record = holding_notes_from_bib_record(mms_id)
+
+      # Get list of holding ids not included in bib record AVA fields
+      missing_holding_ids = all_holding_ids - notes_from_bib_record.keys
+
+      # Bib record AVA fields do not include the holding ID for temporary holding locations.
+      # We have to fetch notes for these holdings individually using their 866
+      # value.
+      notes_from_holdings = holding_notes_from_holding_records(mms_id, missing_holding_ids)
```

Comment:
So if missing_holding_ids is [] there aren't any extra api calls. I remember in the call we talked about checking whether there were temporary locations. Did you discover that wasn't necessary because the code already has to get the holding ids first? That's what it looks like, just wanting to confirm that we're sure we're not making more api calls than we have to!

Reply:
That's correct. The code here tries to get all of the notes it can from the AVA fields on the bib-record. Temporary holding locations will have an AVA entry, but no note and no subfield 8, so those are skipped. The full list of holding ids is already known (from the items API call), the holdings with AVA fields are subtracted from this list, and individual calls to each holding to get the note from their 866 are done only on the remaining ids. It should be the minimum number of API calls needed to get all of the notes.